apps/files/src/store/actions.js

```diff
@@ -215,22 +216,16 @@ function _buildShare (s) {
     case ('6'):
     // fall through
     case ('1'): // group share
-      share.role = 'custom'
-      if (s.permissions === '1' || s.permissions === '17') {
-        share.role = 'viewer'
-      }
-      if (s.permissions === '3' || s.permissions === '15' || s.permissions === '19' || s.permissions === '31') {
-        share.role = 'editor'
-      }
+      share.role = bitmaskToRole(s.permissions, file.type === 'folder')
```

Comment:
if we inject the "roles" object as discussed above, the `bitmaskToRole` does not need to know any more if folder or not because said logic would be done by the caller.

Reply:
=> later